server/service/src/apis/patient_v4.rs

```diff
@@ -0,0 +1,195 @@
+use chrono::NaiveDate;
+use reqwest::{Client, Url};
+use serde::{Deserialize, Deserializer, Serialize};
+
+pub struct PatientApiV4 {
+    server_url: Url,
+    client: Client,
+}
+
+pub struct PatientParamsV4 {
+    pub limit: Option<u32>,
+    pub offset: Option<u32>,
+    pub first_name: Option<String>,
+    pub last_name: Option<String>,
+    pub dob: Option<NaiveDate>,
+    pub policy_number: Option<String>,
+    pub barcode: Option<String>,
+    pub is_deleted: Option<bool>,
+    pub code: Option<String>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct PatientInsurance {
```

Comment:
I removed it, but I don't see the point. This file is a lib for the v4 API. Why would you have it incomplete? Its very little work to create the full type especially if you don't know yet which fields you will need later. It's also just more work when you get back to it because you need more fields, e.g. had this a couple of times for sync translations. Then you have to look at 4d or get it from a Postman request again...

Reply:
> Its very little work to create the full type especially if you don't know yet which fields you will need later.

If code is there it would need to be tested, and then we need to test all the variations of that v4 API type 

> Then you have to look at 4d or get it from a Postman request again...

If there are manual tests for this, it should be very easy (since manual tests would already have the Postman request result ?)

> I removed it, but I don't see the point

I think I can understand what you mean, but surely you must have made a decision that there is `no point` to integrate all of the fields in the actual postman return (you've only made variants for some fields in the patient) ?

```
    {
        "ID": "D1EAEE433FE3440AB4C7A1EBA7AF468F",
        "name": "Check, Check",
        "fax": "",
        "phone": "",
        "customer": true,
        "bill_address1": "",
        "bill_address2": "",
        "supplier": false,
        "charge code": "abc",
        "margin": 0,
        "comment": "",
        "currency_ID": "8F249548D2D54984A52D2E578C9081A8",
        "country": "",
        "freightfac": 1,
        "email": "",
        "custom1": "",
        "code": "abc",
        "last": "Check",
        "first": "Check",
        "title": "",
        "female": false,
        "date_of_birth": "1998-12-12T00:00:00",
        "overpayment": 0,
        "group_ID": "",
        "hold": false,
        "ship_address1": "",
        "ship_address2": "",
        "url": "",
        "barcode": "*abc*",
        "postal_address1": "",
        "postal_address2": "",
        "category1_ID": "",
        "region_ID": "",
        "type": "patient",
        "price_category": "A",
        "flag": "",
        "manufacturer": false,
        "print_invoice_alphabetical": false,
        "custom2": "",
        "custom3": "",
        "default_order_days": 0,
        "connection_type": 0,
        "PATIENT_PHOTO": "[object Picture]",
        "NEXT_OF_KIN_ID": "",
        "POBOX": "",
        "ZIP": 0,
        "middle": "",
        "preferred": false,
        "Blood_Group": "",
        "marital_status": "",
        "Benchmark": false,
        "next_of_kin_relative": "",
        "mother_id": "",
        "postal_address3": "",
        "postal_address4": "",
        "bill_address3": "",
        "bill_address4": "",
        "ship_address3": "",
        "ship_address4": "",
        "ethnicity_ID": "",
        "occupation_ID": "",
        "religion_ID": "",
        "national_health_number": "",
        "Master_RTM_Supplier_Code": 0,
        "ordering_method": "",
        "donor": false,
        "latitude": 0,
        "longitude": 0,
        "Master_RTM_Supplier_name": "",
        "category2_ID": "",
        "category3_ID": "",
        "category4_ID": "",
        "category5_ID": "",
        "category6_ID": "",
        "bill_address5": "",
        "bill_postal_zip_code": "",
        "postal_address5": "",
        "postal_zip_code": "",
        "ship_address5": "",
        "ship_postal_zip_code": "",
        "supplying_store_id": "736928AE81CB4757A6164F732D6A5D00",
        "license_number": "",
        "license_expiry": "0000-00-00",
        "has_current_license": false,
        "custom_data": null,
        "maximum_credit": 0,
        "nationality_ID": "",
        "created_date": "2023-06-19",
        "integration_ID": "",
        "isDeceased": false,
        "is_deleted": false,
        "om_created_datetime": "",
        "om_gender": "",
        "HSH_name": "",
        "HSH_code": "",
        "HSH_ID": "",
        "nameNotes": [],
        "nameInsuranceJoin": []
    }
```

Reply:
Hehe not sure where it got the other version from or if I did some filtering...